spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -200,17 +201,29 @@ private DispatchQueryResponse handleSQLQuery(DispatchQueryRequest dispatchQueryR
     }
   }
 
+  // TODO: Revisit this logic.
+  // Currently, Spark if datasource is not provided in query.
+  // Spark Assumes the datasource to be catalog.
+  // This is required to handle drop index case properly when datasource name is not provided.
+  private static void fillMissingDetails(
```

Comment:
@dai-chen  Incase if I am missing datasource name in any of the index queries. I am adding the datasource name for request.
Currently
`select * from default.http_logs is also working.`

Reply:
chen, could we avoid this logic in plugin, and handle it in spark?

Reply:
@vamsi-amazon , datasource is mandatory field of POST async_query, right?

Reply:
@vamsi-amazon @penghuo I think Flint extensions will qualify all table/mv names.